### DISCUSSION TITLE
Use without SSR

### DISCUSSION THREAD

**Discussion by simplenotezy** [nextauthjs/next-auth#2687] [Help] [answered]
We have a separate API that handles all data, as well as user creation etc. We're planning to deploy our Next.js app as a SPA / static site, and don't plan to have any server running for the backend.

Is it possible to use next-auth without having a Next server running?

**Comment by balazsorban44** [comment, marked as answer]
SSR means server-side rendering, and I'll assume this is not what you actually meant.

If you talk about [static export](https://nextjs.org/docs/advanced-features/static-html-export), then no, it is not. NextAuth.js relies on a secure backend being available at runtime, like Next.js API routes. 

There are plans to support other frameworks #2294, and someone already got `next-auth` working with Express-like backends. Check the issue for more info.

**Reply by simplenotezy** [reply]
I actually meant SSR, since we're statically exporting. But yes, what I was specifically asking was implementing next-auth into another backend instead of relying on Next.js - and we do use Express.js so I'll check the issue you linked to out.

**Reply by simplenotezy** [reply]
In the future it would be great if I had the possibility to mark the answer I believe is correct as "selected". It surprised me you selected your answer as the "only correct one", without first confirming with me ;)

Anyhow, thanks for a prompt reply.

**Reply by balazsorban44** [reply]
SSR does not equal statically exporting, at least not in common sense. I understand that static export will be "rendered by a server", but it will happen build-time, and SSR in my experience is a term used in runtime applications.

I would rather call your concept Static Site Generation, or SSG. I agree that all these acronyms might be confusing. 😬

NextAuth.js (as any other similar library) relies on a safe runtime, as you MUST check if the user is authenticated "on-the-fly". You cannot "statically generate a user authentication".

The question you asked was "Is it possible to use next-auth without having a Next server running?" to which my answer was "No", hence I marked it as answered. :slightly_smiling_face:

**Reply by simplenotezy** [reply]
@balazsorban44 Hey again ;) The thing is, in our project the Next.js project is meant to be frontend only. We don't want to run authentication through Next.js at all. I'd like that part to be handled by our "real" backend. So I'm perfectly aware that SSR is not equal to statically exporting. We want to statically export our Next.js application code, thus eliminating any Next.js API there might be. So as I stated in the subject, what I want to know if Next.js can be run **without SSR**. And it seems from what you've shared that there's a few people who have managed to implement the authentication that's provided by Next-Auth on a Node.js/express server, which is exactly what we're looking into.

Again, we wish not to use the SSR part of Next.js, only the SSG/static site, and handle the authentication on another server. Currently the only library that is dependent on "/api/" for our project is Next-Auth.

Hope it makes sense ☺️

**Reply by balazsorban44** [reply]
Yes, it does. I know for a fact that Express-like backends are already possible, with very minor changes. Here is a pointer https://github.com/s-kris/vite-ssr-starter/pull/1

I am not sure though how it would work if the backend is on a different domain since we are using same-site cookies. If you manage to keep your backend on the same domain, then no issues, I think.

https://github.com/nextauthjs/next-auth/blob/main/src/client/index.js will need some work as well, but possible. Currently, the client always expects `/api/auth` as path. https://github.com/nextauthjs/next-auth/issues/1713. I would like to start working on that (and other things) as soon as v4 is stable.

I hope that helps! Interesting to see our expansion outside Next.js...
